build-system/tasks/performance/helpers.js

```diff
@@ -18,7 +18,16 @@ const fetch = require('node-fetch');
 const fs = require('fs');
 const path = require('path');
 
+const EXTRA_URL_PARAM = {
+  'analytics': 'amp-analytics-performance-param',
+};
+const ANALYTICS_PARAM = Object.keys(EXTRA_URL_PARAM)
+  .map((key) => `${key}=${EXTRA_URL_PARAM[key]}`)
+  .toString();
 const CDN_URL = 'https://cdn.ampproject.org/';
+const AMP_JS_PATH = '/dist/amp.js';
+const LOCAL_PATH_REGEXP = /dist\/(v0\/amp-[A-Za-z\-0-9\.]+).max(.js)/;
```

Comment:
Originally, we thought that `gulp --cdn` to run local pages would work, since that changes the script tags to `cdn.ampproject....` and we can parse out the filepath. 

However, to use `gulp --cdn`, we have to run `gulp dist --fortesting`, which then causes `amp-analytics/config.js` to not work correctly; it requests `file://...v0/analytics-vendors/gtag.json`, which is blocked because AMP doesn't allow requests using the file protocol (it uses the file protocol because `this.win.location.protocol` is file on puppeteer.

So the fix is to use neither gulp --cdn to serve local pages, nor gulp --fortesting, but just parse out the filepaths manually. Also the change wasn't too bad.

Reply:
> But this doesn't explain why you need .max.js and also amp.js, they should never appear when you run gulp dist.

Turns out I was making mistakes when running the `gulp dist --forstesting` + `gulp serve --compiled`.